File: tests/playwright-ts/src/clients/test-data-client.ts
import type { APIRequestContext, APIResponse, TestInfo } from '@playwright/test';
import { Endpoints } from '../core/endpoints.js';
import { BaseApiClient } from './base-api-client.js';

export class TestDataClient extends BaseApiClient {
  constructor(request: APIRequestContext, testInfo: TestInfo) {
    super(request, testInfo);
  }

  async resetData(): Promise<APIResponse> {
    await this.attachRequest('Reset Test Data', 'POST', Endpoints.testReset);

    const response = await this.request.post(Endpoints.testReset);

    await this.attachResponse('Reset Test Data', response);
    return response;
  }
}

File: tests/rest-assured-java/src/test/java/com/portfolio/qa/clients/TestDataClient.java
package com.portfolio.qa.clients;

import com.portfolio.qa.endpoints.Endpoints;
import io.restassured.response.Response;

public class TestDataClient extends BaseApiClient {

    public Response resetData() {
        return request()
            .when()
            .post(Endpoints.TEST_RESET);
    }
}